### PR TITLE
[NUI-1679] add e2e non multipart upload, add better error handling for missing credentials

### PR DESCRIPTION
**Pull request by jdelbick**
## Description

Fixes # https://jira.corp.adobe.com/browse/NUI-1679
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jdelbick** [comment]
testing:
```
357 passing (22s)
  1 pending
  1 failing

  1) Block Download
       Block download small png file:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected ... Lines skipped

  {
    error: [],
...
        sourceFile: '/path/to/image-file-1.png',
        sourceFolder: '/path/to',
+       targetFile: '/Users/jamiedelbick/work/adobe/node-httptransfer/test/tmp.png',
+       targetFolder: '/Users/jamiedelbick/work/adobe/node-httptransfer/test'
-       targetFile: '/Users/mathern/Desktop/git/_adobe/node-httptransfer/test/tmp.png',
-       targetFolder: '/Users/mathern/Desktop/git/_adobe/node-httptransfer/test'
      }
    ],
...
        sourceFile: '/path/to/image-file-1.png',
        sourceFolder: '/path/to',
+       targetFile: '/Users/jamiedelbick/work/adobe/node-httptransfer/test/tmp.png',
+       targetFolder: '/Users/jamiedelbick/work/adobe/node-httptransfer/test',
-       targetFile: '/Users/mathern/Desktop/git/_adobe/node-httptransfer/test/tmp.png',
-       targetFolder: '/Users/mathern/Desktop/git/_adobe/node-httptransfer/test',
        transferred: 1911
      }
...
        sourceFile: '/path/to/image-file-1.png',
        sourceFolder: '/path/to',
+       targetFile: '/Users/jamiedelbick/work/adobe/node-httptransfer/test/tmp.png',
+       targetFolder: '/Users/jamiedelbick/work/adobe/node-httptransfer/test'
-       targetFile: '/Users/mathern/Desktop/git/_adobe/node-httptransfer/test/tmp.png',
-       targetFolder: '/Users/mathern/Desktop/git/_adobe/node-httptransfer/test'
      }
    ]
  }
      + expected - actual

             "fileSize": 1911
             "mimeType": "image/png"
             "sourceFile": "/path/to/image-file-1.png"
             "sourceFolder": "/path/to"
      -      "targetFile": "/Users/jamiedelbick/work/adobe/node-httptransfer/test/tmp.png"
      -      "targetFolder": "/Users/jamiedelbick/work/adobe/node-httptransfer/test"
      +      "targetFile": "/Users/mathern/Desktop/git/_adobe/node-httptransfer/test/tmp.png"
      +      "targetFolder": "/Users/mathern/Desktop/git/_adobe/node-httptransfer/test"
           }
         ]
         "fileprogress": [
           {
--
             "fileSize": 1911
             "mimeType": "image/png"
             "sourceFile": "/path/to/image-file-1.png"
             "sourceFolder": "/path/to"
      -      "targetFile": "/Users/jamiedelbick/work/adobe/node-httptransfer/test/tmp.png"
      -      "targetFolder": "/Users/jamiedelbick/work/adobe/node-httptransfer/test"
      +      "targetFile": "/Users/mathern/Desktop/git/_adobe/node-httptransfer/test/tmp.png"
      +      "targetFolder": "/Users/mathern/Desktop/git/_adobe/node-httptransfer/test"
             "transferred": 1911
           }
         ]
         "filestart": [
--
             "fileSize": 1911
             "mimeType": "image/png"
             "sourceFile": "/path/to/image-file-1.png"
             "sourceFolder": "/path/to"
      -      "targetFile": "/Users/jamiedelbick/work/adobe/node-httptransfer/test/tmp.png"
      -      "targetFolder": "/Users/jamiedelbick/work/adobe/node-httptransfer/test"
      +      "targetFile": "/Users/mathern/Desktop/git/_adobe/node-httptransfer/test/tmp.png"
      +      "targetFolder": "/Users/mathern/Desktop/git/_adobe/node-httptransfer/test"
           }
         ]
       }
      
      at Context.<anonymous> (test/block/blockdownload.test.js:347:16)



---------------------------|---------|----------|---------|---------|------------------------------------------
File                       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                        
---------------------------|---------|----------|---------|---------|------------------------------------------
All files                  |   93.47 |    87.76 |    96.3 |   93.45 |                                          
 lib                       |      96 |    91.15 |   99.14 |      96 |                                          
  aemmultipart.js          |     100 |      100 |     100 |     100 |                                          
  aempartsize.js           |     100 |      100 |     100 |     100 |                                          
  constants.js             |     100 |      100 |     100 |     100 |                                          
  error.js                 |     100 |      100 |     100 |     100 |                                          
  fetch.js                 |   98.24 |    92.85 |     100 |   98.24 | 95                                       
  file.js                  |    89.7 |    84.44 |    92.3 |    89.7 | 46,70,77,111-148                         
  filehandle.js            |   86.04 |    61.11 |     100 |   86.04 | 75,95,97,115,137,159                     
  headers.js               |     100 |    95.12 |     100 |     100 | 90,124                                   
  logger.js                |     100 |      100 |     100 |     100 |                                          
  randomfileaccess.js      |    92.3 |    81.81 |     100 |    92.3 | 52-53,79                                 
  retry.js                 |     100 |      100 |     100 |     100 |                                          
  stream.js                |     100 |       90 |     100 |     100 | 123                                      
  util.js                  |   96.61 |    88.88 |     100 |   96.61 | 32,94                                    
 lib/aem                   |     100 |       95 |     100 |     100 |                                          
  aemdownload.js           |     100 |      100 |     100 |     100 |                                          
  aemupload.js             |     100 |    91.66 |     100 |     100 | 121                                      
 lib/asset                 |   97.94 |    96.89 |     100 |   97.94 |                                          
  asset.js                 |     100 |      100 |     100 |     100 |                                          
  assetmetadata.js         |     100 |      100 |     100 |     100 |                                          
  assetmultipart.js        |     100 |      100 |     100 |     100 |                                          
  assetversion.js          |     100 |      100 |     100 |     100 |                                          
  nameconflictpolicy.js    |     100 |      100 |     100 |     100 |                                          
  transferasset.js         |     100 |    98.11 |     100 |     100 | 240                                      
  transferpart.js          |   84.21 |     62.5 |     100 |   84.21 | 38,41,44                                 
 lib/block                 |   85.71 |    70.37 |   83.33 |   85.71 |                                          
  blockdownload.js         |   97.64 |    66.66 |     100 |   97.64 | 153,166                                  
  blockupload.js           |   98.05 |     82.5 |     100 |   98.05 | 198,211                                  
  download-error.js        |    36.5 |    32.65 |   33.33 |    36.5 | 27,55,62,64,66,68,71-74,80,87-95,121-193 
  error-codes.js           |     100 |      100 |     100 |     100 |                                          
  upload-error.js          |   98.41 |      100 |     100 |   98.41 | 95                                       
 lib/controller            |   75.43 |    68.18 |   76.92 |   75.43 |                                          
  transfercontroller.js    |   78.94 |     62.5 |   85.71 |   78.94 | 45,202,211-223                           
  transferevent.js         |   68.42 |    71.42 |   66.66 |   68.42 | 36,39,42,45,62,80                        
 lib/functions             |   95.42 |    93.38 |     100 |   95.39 |                                          
  aemcompleteupload.js     |     100 |      100 |     100 |     100 |                                          
  aeminitiateupload.js     |     100 |      100 |     100 |     100 |                                          
  closefiles.js            |     100 |      100 |     100 |     100 |                                          
  failunsupportedassets.js |     100 |      100 |     100 |     100 |                                          
  filterfailedassets.js    |     100 |      100 |     100 |     100 |                                          
  getassetmetadata.js      |      75 |       50 |     100 |      75 | 66-68,70-75                              
  transfer.js              |   94.54 |    89.47 |     100 |   94.54 | 65,69,101                                
  transferpartscreate.js   |     100 |      100 |     100 |     100 |                                          
  transferpartsjoin.js     |   96.15 |      100 |     100 |   96.15 | 86                                       
 lib/generator             |     100 |      100 |     100 |     100 |                                          
  function.js              |     100 |      100 |     100 |     100 |                                          
  mapconcurrent.js         |     100 |      100 |     100 |     100 |                                          
  pipeline.js              |     100 |      100 |     100 |     100 |                                          
---------------------------|---------|----------|---------|---------|------------------------------------------
```
that one download test is still failing for me due to hardcoded filepath. @tmathern are you already in progress on fixing this or should I?

**Comment by jdelbick** [comment]
weird that unit tests (github actions) still are not running here...

File: README.md [hunk 1]
```diff
@@ -134,6 +134,17 @@ with an Adobe Experience Manager instance. To run the tests:
 * Create a `.env` file by following the instructions in [.env_example](./e2e/.env_example).
 * Run the tests by executing `npm run e2e` from the root directory of the repository.
 
+## End-to-End Block upload/download Tests
```
Comment (tmathern) [inline README.md:137]:
```suggestion
## End-to-End block upload/download Tests
```

**Review by tmathern — APPROVED** [review]

**Comment by tmathern** [comment]
btw, unit test failure is fixed in https://github.com/adobe/node-httptransfer/pull/81

**Pull request merged**